pkg/kv/kvserver/rangefeed/catchup_scan.go

```diff
@@ -175,213 +176,216 @@ func (i *CatchUpIterator) CatchUpScan(
 		} else if !ok {
 			break
 		}
-
-		if err := i.pacer.Pace(ctx); err != nil {
-			// We're unable to pace things automatically -- shout loudly
-			// semi-infrequently but don't fail the rangefeed itself.
-			if every.ShouldLog() {
-				log.Errorf(ctx, "automatic pacing: %v", err)
+		select {
+		case <-streamCtx.Done():
+			return streamCtx.Err()
+		default:
```

Comment:
We might want to run the microbenchmarks we have for this file.  An alternative would be to put this into outputEvents() above so that we would just call it on a per-flush basis.  Although that kinda raises the question of why the send function doesn't take a context. But that is likely a larger refactoring of the code.